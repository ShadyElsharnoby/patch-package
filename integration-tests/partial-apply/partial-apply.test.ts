import { runIntegrationTest } from "../runIntegrationTest"
runIntegrationTest({
  projectName: "partial-apply",
  shouldProduceSnapshots: true,
})
