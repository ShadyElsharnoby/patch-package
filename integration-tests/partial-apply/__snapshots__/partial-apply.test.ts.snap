// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test partial-apply: 00: patch-package fails when one of the patches in the sequence fails 1`] = `
"SNAPSHOT: patch-package fails when one of the patches in the sequence fails
patch-package 0.0.0
Applying patches...
left-pad@1.3.0 (1 hello) ✔

⛔ ERROR

Failed to apply patch file left-pad+1.3.0+002+world.patch.

If this patch file is no longer useful, delete it and run

  patch-package

To partially apply the patch (if possible) and output a log of errors to fix, run

  patch-package --partial

After which you should make any required changes inside node_modules/left-pad, and finally run

  patch-package left-pad

to update the patch file.

END SNAPSHOT"
`;

exports[`Test partial-apply: 01: patch-package --partial saves a log 1`] = `
"SNAPSHOT: patch-package --partial saves a log
patch-package 0.0.0
Applying patches...
left-pad@1.3.0 (1 hello) ✔
Saving errors to ./patch-package-errors.log
patch-package-errors.log
Cannot apply hunk 0 for file node_modules/left-pad/index.js
\`\`\`diff
@@ -3,7 +3,7 @@
      * and/or modify it under the terms of the Do What The Fuck You Want
      * To Public License, Version 2, as published by Sam Hocevar. See
      * http://www.wtfpl.net/ for more details. */
-'use oops';
+'use world';
 module.exports = leftPad;
 
 var cache = [
\`\`\`
END SNAPSHOT"
`;
